config/routes.rb

```diff
@@ -16,6 +16,9 @@
         get 'files/*filepath', to: 'files#show', format: false, as: "file"
         resources :submissions, only: %i[create]
       end
+      resources :submission, only: [] do
+        resources :cancellations, only: %i[create], controller: "submissions/cancellations"
+      end
```

Comment:
I'll change this to be a method on submissions in a separate PR shortly, so as not to block :)